huaweicloud/services/acceptance/vpn/resource_huaweicloud_vpn_customer_gateway_test.go

```diff
@@ -88,11 +88,81 @@ func TestAccCustomerGateway_basic(t *testing.T) {
 	})
 }
 
+func TestAccCustomerGateway_certificate(t *testing.T) {
+	var obj interface{}
+
+	name := acceptance.RandomAccResourceName()
+	rName := "huaweicloud_vpn_customer_gateway.test"
+	ipAddress := "172.16.2.1"
+	certificateContent := acceptance.HW_CERTIFICATE_CONTENT
+	certificateContentUpdate := acceptance.HW_CERTIFICATE_CONTENT_UPDATE
+
+	rc := acceptance.InitResourceCheck(
+		rName,
+		&obj,
+		getCustomerGatewayResourceFunc,
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acceptance.TestAccPreCheck(t)
+			acceptance.TestAccPreCheckUpdateCertificateContent(t)
+		},
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      rc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testCustomerGateway_certificate(name, ipAddress, certificateContent),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(rName, "name", name),
+					resource.TestCheckResourceAttr(rName, "ip", "172.16.2.1"),
```

Comment:
"172.16.2.1" should be replaced with `ipAddress`
```suggestion
					resource.TestCheckResourceAttr(rName, "ip", ipAddress),
```